DSharpPlus/Entities/Channel/DiscordChannel.cs

```diff
@@ -624,34 +624,72 @@ public async Task<ThreadQueryResult> ListJoinedPrivateArchivedThreadsAsync(DateT
     }
 
     /// <summary>
-    /// Deletes multiple messages if they are less than 14 days old.  If they are older, none of the messages will be deleted and you will receive a <see cref="BadRequestException"/> error.
+    /// Deletes multiple messages if they are less than 14 days old.  If they are older, none of the messages will be deleted and you will receive a <see cref="ArgumentException"/> error.
     /// </summary>
     /// <param name="messages">A collection of messages to delete.</param>
     /// <param name="reason">Reason for audit logs.</param>
-    /// <returns></returns>
-    /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ManageMessages"/> permission.</exception>
-    /// <exception cref="NotFoundException">Thrown when the channel does not exist.</exception>
-    /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
-    /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public async Task DeleteMessagesAsync(IEnumerable<DiscordMessage> messages, string reason = null)
+    /// <remarks>One api call per 100 requests</remarks>
+    public async Task<int> DeleteMessagesAsync(IReadOnlyList<DiscordMessage> messages, string reason = null)
     {
-        // don't enumerate more than once
-        ulong[] msgs = messages.Where(x => x.Channel.Id == this.Id).Select(x => x.Id).ToArray();
-        if (messages == null || !msgs.Any())
+        ArgumentNullException.ThrowIfNull(messages, nameof(messages));
+
+        List<ulong> messagesList = [];
+        foreach (DiscordMessage message in messages)
+        {
+            if (message.ChannelId != this.Id)
+            {
+                throw new ArgumentException(
+                    "You can only delete messages from the channel this channel object is representing.");
+            }
+            else if (message.Timestamp < DateTimeOffset.UtcNow.AddDays(-14))
+            {
+                throw new ArgumentException("You can only delete messages that are less than 14 days old.");
+            }
+
+            messagesList.Add(message.Id);
+        }
+
+        if (messagesList.Count == 0)
         {
             throw new ArgumentException("You need to specify at least one message to delete.");
         }
+        else if (messagesList.Count < 2)
+        {
+            await this.Discord.ApiClient.DeleteMessageAsync(this.Id, messagesList.Single(), reason);
```

Comment:
Any specific reason for `Single` over `First`?

Reply:
dont think it matters, didnt think about it

Reply:
I don't think it genuinely does matter. I just use `First` since it's more specific to the operation I'm doing. This questions was driven more by curiosity.

Reply:
Actually wait it does matter. Index by `0` instead.